pkg/ingester-rf1/metrics.go

```diff
@@ -3,55 +3,61 @@ package ingesterrf1
+	flushSize                         prometheus.Histogram
+	limiterEnabled                    prometheus.Gauge
+	segmentSize                       prometheus.Histogram
+	// Shutdown marker for ingester scale down.
 	shutdownMarker prometheus.Gauge
 }
 
 func newIngesterMetrics(r prometheus.Registerer, metricsNamespace string) *ingesterMetrics {
 	return &ingesterMetrics{
-		limiterEnabled: promauto.With(r).NewGauge(prometheus.GaugeOpts{
-			Name: "loki_ingester_rf1_limiter_enabled",
-			Help: "Whether the ingester's limiter is enabled",
-		}),
 		autoForgetUnhealthyIngestersTotal: promauto.With(r).NewCounter(prometheus.CounterOpts{
 			Name: "loki_ingester_rf1_autoforget_unhealthy_ingesters_total",
 			Help: "Total number of ingesters automatically forgotten",
 		}),
-		chunksFlushFailures: promauto.With(r).NewCounter(prometheus.CounterOpts{
-			Namespace: constants.Loki,
-			Name:      "ingester_rf1_chunks_flush_failures_total",
-			Help:      "Total number of flush failures.",
-		}),
-		chunksFlushedPerReason: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
-			Namespace: constants.Loki,
-			Name:      "ingester_rf1_chunks_flushed_total",
-			Help:      "Total flushed chunks per reason.",
-		}, []string{"reason"}),
-		flushedChunksStats: analytics.NewCounter("ingester_rf1_flushed_chunks"),
-		flushQueueLength: promauto.With(r).NewGauge(prometheus.GaugeOpts{
-			Namespace: metricsNamespace,
-			Subsystem: "ingester_rf1",
-			Name:      "flush_queue_length",
-			Help:      "The total number of series pending in the flush queue.",
+		flushDuration: promauto.With(r).NewHistogram(prometheus.HistogramOpts{
+			Name:    "loki_ingester_rf1_flush_duration",
+			Help:    "The flush duration (in seconds)",
+			Buckets: []float64{0.1, 0.2, 0.5, 1, 2.5, 5, 7.5, 10},
+		}),
+		flushFailuresTotal: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Name: "loki_ingester_rf1_flush_failures_total",
+			Help: "The total number of failed flushes.",
+		}),
+		flushesTotal: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Name: "loki_ingester_rf1_flushes_total",
+			Help: "The total number of flushes.",
+		}),
+		flushQueues: promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Name: "loki_ingester_rf1_flush_queues",
+			Help: "The total number of flush queues.",
+		}),
+		flushSize: promauto.With(r).NewHistogram(prometheus.HistogramOpts{
+			Name:    "loki_ingester_rf1_flush_size_megabytes",
+			Help:    "The flush size (as written to object storage).",
+			Buckets: []float64{0.1, 0.5, 1, 2.5, 5, 7.5, 10},
+		}),
+		limiterEnabled: promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Name: "loki_ingester_rf1_limiter_enabled",
+			Help: "1 if the limiter is enabled, otherwise 0.",
+		}),
+		segmentSize: promauto.With(r).NewHistogram(prometheus.HistogramOpts{
+			Name:    "loki_ingester_rf1_segment_size_megabytes",
+			Help:    "The segment size (uncompressed in memory).",
+			Buckets: []float64{0.1, 0.5, 1, 2.5, 5, 7.5, 10},
```

Comment:
Maybe we want a bigger bucket for the case of large Push generating a very large segment. Especially as our target or ~8Mb is quite close to the top of this range.

Reply:
imo for any new histogram metrics we're creating we should be using native histograms: https://github.com/prometheus/client_golang/blob/45f1e72421d9d11af6be784ad60b7389f7543e70/examples/random/main.go#L51-L67

also consider leaving the metrics themselves at `bytes` as the units, grafana has nice support for dashboard panels to set the units, so you don't have to do the conversion when querying